plugins/zenoh-plugin-storage-manager/src/replica/storage.rs

```diff
@@ -240,7 +240,7 @@ impl StorageService {
                                 continue;
                             }
                         };
-                        let timestamp = sample.timestamp().cloned().unwrap_or(new_timestamp(zid));
+                        let timestamp = sample.timestamp().cloned().unwrap_or(session_timestamp);
```

Comment:
This `session_timestamp` is created only once at storage creation (line 152).
Using it here implies that all sample received later without timestamp will get this same `session_timestamp`.
This could lead to weird behaviour with consolidation on `get` operations: a recent publication will be timestamped with an old time (`session_timestamp`) and being discarded if another queryable has a more recent timestamp.

 A new timestamp shall be generated for each sample received without timestamp.

Reply:
Makes sense. Fixed!